File: max_sub_arr/max_sub_arr.java
public class max_sub_arr {
    public static void main(String[] args){
        int[] nums = {1,2,3,4,5};
        System.out.println(max_sub(nums));
    }
    private static int max_sub(int[] nums){
        int curr_max = 0; int end_max = 0;
        for(int i = 1; i < nums.length; i++){
            end_max = Math.max(nums[i], nums[i] + end_max);
            curr_max = Math.max(curr_max, end_max);
        } return curr_max;
    }
}

File: max_sub_arr/max_sub_arr.py
nums = [-2,1,-3,4,-1,2,1,-5,4]

def solution(nums):
    total = nums[0]
    curr = nums[0]

    for i in range(1, len(nums)):
        total = max(nums[i], total + nums[i])
        curr = max(curr, total)
    return curr

print(solution(nums))